File: name-inverter/name-inverter-java/src/main/java/pl/frognar/nameinverter/_05/NameInverter.java
package pl.frognar.nameinverter._05;

import com.google.common.collect.Lists;

import java.util.ArrayList;
import java.util.List;

public class NameInverter {
    public static String invert(String name) {
        List<String> nameParts = breakIntoPartsIgnoringWhitespaces(name);
        if (nameParts.size() < 2)
            return nameParts.get(0);
        String first = nameParts.get(0);
        String last = nameParts.get(1);
        return String.format("%s, %s", last, first);
    }

    private static ArrayList<String> breakIntoPartsIgnoringWhitespaces(String name) {
        return Lists.newArrayList(name.trim().split(RegularExpressions.anyAmountOfWhitespaces()));
    }
}

class RegularExpressions {
    public static String anyAmountOfWhitespaces() {
        return "\\s+";
    }
}

File: name-inverter/name-inverter-py/03/NameInverter.py
import re
from typing import List


class NameInverter:
    @classmethod
    def invert(cls, name: str) -> str:
        name_parts = cls.break_into_parts(name)
        if len(name_parts) < 2:
            return name_parts[0]
        return cls.invert_parts(name_parts)

    @classmethod
    def break_into_parts(cls, name: str) -> List[str]:
        return re.split(cls.any_amount_of_whitespaces(), name.strip())

    @classmethod
    def any_amount_of_whitespaces(cls) -> str:
        return '\\s+'

    @classmethod
    def invert_parts(cls, name_parts: List[str]) -> str:
        first = name_parts[0]
        last = name_parts[1]
        return f'{last}, {first}'
